front/app/containers/EventsPage/EventsViewer/index.tsx

```diff
@@ -118,7 +141,45 @@ const EventsViewer = ({
     sort: eventsTime === 'past' ? 'start_at' : '-start_at',
     pageNumber: currentPage,
     attendeeId,
+    ongoing_during: ongoingDuringDates,
   });
+
+  useEffect(() => {
+    if (projectId) {
+      setProjectIdList([projectId]);
+    }
+  }, [projectId]);
+
+  // Update projectIds URL params based on state
+  useEffect(() => {
+    const hasProjectFilter = projectIdList?.length;
+    if (!location.pathname.includes('/projects')) {
+      if (eventsTime === 'past') {
+        updateSearchParams({
+          past_events_project_ids: hasProjectFilter ? projectIdList : null,
+        });
+      } else if (eventsTime === 'currentAndFuture') {
+        updateSearchParams({
+          ongoing_events_project_ids: hasProjectFilter ? projectIdList : null,
+        });
+      }
+    }
+  }, [eventsTime, projectIdList]);
```

Comment:
It looks like the check in the if condition determines only the key to be used. If that is the case, we can change it to. 
```suggestion
useEffect(() => {
  const hasProjectFilter = projectIdList?.length;
  const eventParam = eventsTime === 'past' ? 'past_events_project_ids' : 'ongoing_events_project_ids';

  if (!location.pathname.includes('/projects')) {
    updateSearchParams({
      [eventParam]: hasProjectFilter ? projectIdList : null,
    });
  }
}, [eventsTime, projectIdList]);
```